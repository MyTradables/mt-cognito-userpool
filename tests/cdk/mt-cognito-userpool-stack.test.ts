import { SynthUtils } from '@aws-cdk/assert';
import { App } from '@aws-cdk/core';
import environmentConfig from '../../bin/stack-config';
import MtCognitoUserpoolStack from '../../lib/mt-cognito-userpool-stack';

const app = new App();

function fixGeneratedCdkAssetName() {
  // CDK snapshot contains Lambda code as assets, and asset name is being regenerated every time, causing false negative test results
  // code below replaces regenerated part of assets name with "[HASH REMOVED]"
  expect.addSnapshotSerializer({
    test: (val) => typeof val === 'string' && val.match(/AssetParameters([A-Fa-f0-9]{64})(\w+)/) != null,
    print: () => '"AssetParameters-[HASH REMOVED]"',
  });

  expect.addSnapshotSerializer({
    test: (val) => typeof val === 'string' && val.match(/(\w+) for asset\s?(version)?\s?"([A-Fa-f0-9]{64})"/) != null,
    print: (val) => `"${(val as string).replace(/(\w+ for asset)\s?(version)?\s?"([A-Fa-f0-9]{64})"/, '$1 [HASH REMOVED]')}"`,
  });

  expect.addSnapshotSerializer({
    test: (val) => typeof val === 'number' && val > 1000000000,
    print: () => '[HASH REMOVED]',
  });
}

fixGeneratedCdkAssetName();

describe('Stack validation', () => {
  it('should have a correct stack configuration', () => {
    const stack = new MtCognitoUserpoolStack(app, 'mt-cognito-userpool', environmentConfig);
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });
});