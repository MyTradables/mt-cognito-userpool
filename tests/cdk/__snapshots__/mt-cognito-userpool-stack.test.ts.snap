// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stack validation should have a correct stack configuration 1`] = `
Object {
  "Outputs": Object {
    "UserpoolClientId": Object {
      "Value": Object {
        "Ref": "mttraderspoolclient2D2D912D",
      },
    },
    "UserpoolId": Object {
      "Value": Object {
        "Ref": "mttraderspool7C3F78E0",
      },
    },
  },
  "Resources": Object {
    "mtfacebookloginDA10E1C5": Object {
      "Properties": Object {
        "AttributeMapping": Object {
          "email": "email",
          "family_name": "last_name",
          "given_name": "first_name",
          "nickname": "name",
        },
        "ProviderDetails": Object {
          "api_version": "",
          "authorize_scopes": "public_profile",
          "client_id": "",
          "client_secret": "",
        },
        "ProviderName": "Facebook",
        "ProviderType": "Facebook",
        "UserPoolId": Object {
          "Ref": "mttraderspool7C3F78E0",
        },
      },
      "Type": "AWS::Cognito::UserPoolIdentityProvider",
    },
    "mtgoogleloginC7136A5C": Object {
      "Properties": Object {
        "AttributeMapping": Object {
          "email": "email",
          "family_name": "family_name",
          "given_name": "given_name",
          "nickname": "name",
        },
        "ProviderDetails": Object {
          "authorize_scopes": "profile",
          "client_id": "",
          "client_secret": "",
        },
        "ProviderName": "Google",
        "ProviderType": "Google",
        "UserPoolId": Object {
          "Ref": "mttraderspool7C3F78E0",
        },
      },
      "Type": "AWS::Cognito::UserPoolIdentityProvider",
    },
    "mttraderspool7C3F78E0": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "AdminCreateUserConfig": Object {
          "AllowAdminCreateUserOnly": false,
          "InviteMessageTemplate": Object {
            "EmailMessage": "Dear {username},
  Thank you for signing up at www.mytradables.com!
  
  You can login to your account using:
  Username: {username}
  Password: {####}
  
  Cheers!,
  
  www.mytradables.com",
            "EmailSubject": "Congratulations with your MyTradables account!",
          },
        },
        "AutoVerifiedAttributes": Array [
          "email",
        ],
        "EmailConfiguration": Object {
          "From": "mytradables@gmail.com",
        },
        "EmailVerificationMessage": "The verification code to your new account is {####}",
        "EmailVerificationSubject": "Verify your new myTradables account!",
        "Policies": Object {
          "PasswordPolicy": Object {
            "MinimumLength": 12,
            "RequireLowercase": true,
            "RequireNumbers": true,
            "RequireSymbols": true,
            "RequireUppercase": true,
            "TemporaryPasswordValidityDays": 3,
          },
        },
        "SmsVerificationMessage": "The verification code to your new account is {####}",
        "UserPoolName": "mt-traders-pool",
        "VerificationMessageTemplate": Object {
          "DefaultEmailOption": "CONFIRM_WITH_CODE",
          "EmailMessage": "The verification code to your new account is {####}",
          "EmailSubject": "Verify your new myTradables account!",
          "SmsMessage": "The verification code to your new account is {####}",
        },
      },
      "Type": "AWS::Cognito::UserPool",
      "UpdateReplacePolicy": "Retain",
    },
    "mttraderspoolclient2D2D912D": Object {
      "Properties": Object {
        "AccessTokenValidity": 120,
        "AllowedOAuthFlows": Array [
          "implicit",
          "code",
        ],
        "AllowedOAuthFlowsUserPoolClient": true,
        "AllowedOAuthScopes": Array [
          "profile",
          "phone",
          "email",
          "openid",
          "aws.cognito.signin.user.admin",
        ],
        "CallbackURLs": Array [
          "https://example.com",
        ],
        "ClientName": "mt-traders-pool-client",
        "IdTokenValidity": 120,
        "SupportedIdentityProviders": Array [
          "Google",
          "Facebook",
          "COGNITO",
        ],
        "TokenValidityUnits": Object {
          "AccessToken": "minutes",
          "IdToken": "minutes",
        },
        "UserPoolId": Object {
          "Ref": "mttraderspool7C3F78E0",
        },
      },
      "Type": "AWS::Cognito::UserPoolClient",
    },
  },
}
`;
